airbyte-integrations/connectors/source-braze/source_braze/datetime_stream_slicer.py

```diff
@@ -0,0 +1,49 @@
+import datetime
+import operator
+from dataclasses import dataclass
+from typing import Optional, Mapping, Any
+
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters import RequestOption
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOptionType
+from airbyte_cdk.sources.declarative.stream_slicers import DatetimeStreamSlicer
+from airbyte_cdk.sources.declarative.types import StreamSlice
+
+
+@dataclass
+class DatetimeStreamSlicerComponent(DatetimeStreamSlicer):
```

Comment:
can you create an issue requesting that this is added to the datetimestreamslicer [like described here](https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1670428588243999) and clarifying what the new component should do, and add the issue to the docstring?

Reply:
I don't understand why this is different than the datetimestreamslicer. Can you clarify in the docstring? doesn't datetime stream slicer already have step size?

Reply:
Step exists in basic `DatetimeStreamSlicer` but I need it to be an option which will be passed as a request parameter and to be calculated based on difference between start/end dates. I need the step amount to be more than actual size of the difference by 1, but not for fist slice. I also do not need last slice if start/end dates a equal. And it is important thing that step amount is dynamic and because last slice must not be equal to the step amount if the difference lesser than in step. All of this is not a standard thing and I am not sure we need it in `DatetimeStreamSlicer`. 
Why I need it:
Braze API provides us with `ending_at` and `length` query params, so we can to specify latest point and then move back by `length` until reach `start_date`(initially specified by user). Date specified in `ending_at` is an exclusive property which means we are getting `length` records strictly less than `ending_at`, but `DatetimeStreamSlicer` adds 1 day to previous end date to calculate next start date for next slice, that is why we loose 1 record between each slice. Having `step` query parameter amount one higher than actual difference lets us avoid the problem with gaps in slice windows